src/background.js

```diff
@@ -1,29 +1,44 @@
 window.scrNoti = window.scrNoti || {};
+const seenMessages = {};
+
+// TODO:
+// onQuit
+// mark read?
+// Update seenMessages when changing settings: simple/extended and folders
 
 //==========================================
 // On new email...
 //==========================================
 window.scrNoti.newEmailListener = async (folder, messages) => {
-  let hasNotJunk = false;
 
   if (!(await window.scrNoti.isFolderToCheck(folder))) {
     return;
   }
 
+  // Find new messages, which have not been seen yet
   if (messages && messages.messages && messages.messages.length > 0) {
+
+    let sent = false;
+    const { scriptType } = await messenger.storage.local.get({
+      scriptType: "simple",
+    });
+
     for (const message of messages.messages) {
       if (message && !message.junk) {
-        hasNotJunk = true;
-        break;
+        if (!seenMessages[folder.accountId + folder.path].has(message.id)) {
+          if (!sent) {
+            await window.scrNoti.notifyNativeScript(message, "new");
+            if (scriptType == "simple") {
+              sent = true;
+            }
+          }
+          seenMessages[folder.accountId + folder.path].add(message.id);
```

Comment:
I don't see why we still keep track of the seen messages when we use the "simple" mode, but it is not that much of an issue.

Reply:
This is not needed anymore, if the add-on in "simple" mode only sends one message for one or many new emails.

When I started to write my first external script, I was of the impression, that I get a message for _every_ new message found. I then counted the number of calls and could display, how many new messages are waiting for me. Later I found out, that only one single message is sent, even if there are some more new messages. I found that a bit disappointing (and that was the start of the "extended" mode).

Should it be removed for the "simple" mode?

Reply:
Yes, if possible. The "simple" mode only tells if there are unread messages or not. If it's possible to not consume memory for those "seen messages" then, it would be nice.

Reply:
Done.